packages/sanes-chrome-extension/src/routes/tx-request/components/ShowRequest/ReqRegisterUsernameTx.tsx

```diff
@@ -1,16 +1,74 @@
 import { RegisterUsernameTx } from '@iov/bns';
-import { List } from 'medulas-react-components/lib/components/List';
+import { List, ListItem, ListItemText } from 'medulas-react-components/lib/components/List';
 import * as React from 'react';
+import { makeStyles } from '@material-ui/core';
+import { amountToString } from '../../../../utils/balances';
+import Typography from 'medulas-react-components/lib/components/Typography';
+import Block from 'medulas-react-components/lib/components/Block';
+
+export const REQ_REGISTER_USERNAME = 'req-register-username-tx';
+
+const useStyles = makeStyles({
+  root: {
+    margin: 0,
+  },
+});
+
+const secondaryProps = {
+  noWrap: true,
+};
 
 interface Props {
   readonly creator: string;
   readonly tx: RegisterUsernameTx;
 }
 
-const ReqRegisterUsernameTx = ({ tx, creator }: Props): JSX.Element => {
-  // TODO: implement
-  // https://github.com/iov-one/ponferrada/issues/349
-  return <List />;
+const ReqRegisterUsernameTx = ({ tx }: Props): JSX.Element => {
+  const classes = useStyles();
+
+  const listItemClasses = { root: classes.root };
+
+  const txFee =
+    tx.fee && tx.fee.tokens ? (
+      <React.Fragment>
+        <Block marginTop={1} />
+        <Typography variant="body1" inline>
+          Fee:{' '}
+        </Typography>
+        <Typography variant="body1" inline>
+          {amountToString(tx.fee.tokens)}
+        </Typography>
+      </React.Fragment>
+    ) : (
+      undefined
+    );
```

Comment:
Nice. Can we deduplicate this code block to use th same code in `ReqRegisterUsernameTx` and `ReqSendTransaction`?

Reply:
I used different code for transaction fee in username as opposed to the send transaction. Because I used `List` component here to show addresses only. Take a look at screenshot. Anyway I can use same approach for send transaction too.

Reply:
That would be very useful, since very soon we need to be able to create large amounts of those components quickly. The more we can deduplicate, the better.

Reply:
Okay, now that the code is very similar for RegisterUsername and Send, can we have a separate `TransationFee` component, that just takes a `fee?: Fee` property, where type `Fee` is from `@iov/bcp`?

Such that we just write

```
<TransationFee fee={tx.fee} />
```

in this file and in ReqSendTransaction.tsx.

In the case that `tx.fee` is undefined, this can just be an empty compnent. No need to optimize for this case since it will not happen in reality